symengine/expression.h

```diff
@@ -32,16 +32,16 @@ class Expression
     //! Construct Expression from `int`
     Expression(int n) : m_basic(integer(n)) {}
     //! Construct Expression from Basic
-    template <typename T, typename = typename std::enable_if<std::is_constructible<RCP<const Basic>, T &&>::value>::type>
+    template <typename T>
```

Comment:
I used 15.0.3. I am a bit surprised it doesn't work. I can try again. Does Piranha compile with the latest icpc?

I like that Expression could be used to write expressions in a natural way. We just need to use if it still works if we use `RCP<const Basic>` directly instead of the template. Or I just need to figure out how to make it work with the Intel compiler.

I am ok supporting just the very latest Intel compiler, that's fine.